### ISSUE TITLE
It's impossible to create union with single element using new `Annotated` syntax

### ISSUE THREAD

**Issue by ThirVondukr** [strawberry-graphql/strawberry#2902]
Creating union with a single element using annotated produces a type in schema instead of a union object:
```py
@strawberry.type
class User:
    name: str
    age: int


OneUnion = Annotated[Union[User], strawberry.union(name="OneUnion")]

@strawberry.type
class Query:
    @strawberry.field
    def one(self) -> OneUnion:
        ...
```
```graphql
type Query {
  one: User!
}

type User {
  name: String!
  age: Int!
}
```

If `strawberry.union` is used directly it behaves as expected:
```py
OneUnionOld = strawberry.union(name="OneUnionOld", types=(User,))
```
```gql
union OneUnionOld = User

type Query {
  oneOld: OneUnionOld!
}

type User {
  name: String!
  age: Int!
}
```

Playground link: https://play.strawberry.rocks/?gist=6b1d5a952c2a6348506457782642d3cf

**Comment by patrick91** [comment]
@ThirVondukr ouch, this might be something that needs a workaround:

<img width="657" alt="CleanShot 2023-06-29 at 10 40 49@2x" src="https://github.com/strawberry-graphql/strawberry/assets/667029/36c68762-388e-4d79-80e9-2c8966f74ca7">

**Comment by patrick91** [comment]
I guess we have two options, one we can check if a single type has annotated strawberry.union
and the second option is to have a placeholder type so you can do `Union[YourType, StrawberryPlaceholder]`

**Comment by ThirVondukr** [comment]
@patrick91 Ellipsis `...` could act as a placeholder too, also can't you inspect params in `Annotated` and see if first parameter is a union or there's a `strawberry.union` in args somewhere?

**Comment by patrick91** [comment]
good idea for using `...`, we just need to check if it works with pyright and mypy 🗡️ 

> also can't you inspect params in Annotated and see if first parameter is a union or there's a strawberry.union in args somewhere?

yup, that's the first option 😊

**Comment by ThirVondukr** [comment]
IMO it would be better to go with inspecting args since `Annotated[User, strawberry.union(...)]` would be valid this way too, also `Union[User, ...]` seems weird

**Comment by patrick91** [comment]
Yup, I agree, I pointed the other versions just in case we can't do the fist 😊

**Comment by patrick91** [comment]
This has been fixed in #2982
